server/util/strategicLogger.js

```diff
@@ -0,0 +1,176 @@
+  const date = contrib.created_at;
+  const alias = (await getAmazonAliasByGithub(pg, contrib.user.login))
+    .amazon_alias;
+  const githubStat = contrib.state;
+  const contribUrl = contrib.html_url;
+  const githubId = contrib.id;
+
+  const approver = 1; // get approverId
+  const approvalStat = 'approved-strategic';
+  return await pg.none(
+    'insert into contributions (project_id, contribution_description, contribution_date, ' +
+      'contributor_alias, contribution_github_status, contribution_url, github_id, approver_id, ' +
+      'approval_status, approval_notes, approval_date, contribution_submission_date, ' +
+      'contribution_closed_date, contribution_project_review, contribution_metadata) ' +
+      'values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15:json)',
+    [
+      projId,
+      desc,
+      date,
+      alias,
+      githubStat,
+      contribUrl,
+      githubId,
+      approver,
+      approvalStat,
+      null,
+      null,
+      date,
+      date,
+      true,
+      null,
+    ]
+  );
+}
+
+async function whitelistedContrib(pg, contrib, groups) {
+  const username = contrib.user.login;
+  const user = await listGroupsByGithub(pg, username);
+  if (user && user.groups.length !== 0) {
+    return groups.some(r => user.groups.includes(r));
+  }
+  return false;
+}
+
+async function fetchNewContribs(pg, owner, repo) {
+  let response = await octokit.pullRequests.getAll({
+    owner: owner.toString(),
+    repo: repo.toString(),
+    state: 'all',
+  });
+  let { data } = response;
+
+  let lastContribId = data[data.length - 1].id;
+
+  while (
+    octokit.hasNextPage(response) &&
+    (await contributionExists(pg, lastContribId)).length === 0
+  ) {
+    response = await octokit.getNextPage(response);
+    data = data.concat(response.data);
+    lastContribId = data[data.length - 1].id;
+  }
+
+  return await data;
+}
+
+async function updateStrategicContribs(pg) {
+  const projects = await getAllStrategicProjects(pg);
+  projects.forEach(async proj => {
+    const project = await searchProjectById(pg, parseInt(proj.id, 10));
+    const url = project[0].project_url;
+    const urlTokens = url.split('/');
+    const last = urlTokens.length - 1;
+    const repo = urlTokens[last];
+    const owner = urlTokens[last - 1];
+
+    // groups this project belongs to
+    const groups = (await searchGroupsByProjectId(pg, proj.id)).groups;
+
+    // find all new contributions
+    const contribs = await fetchNewContribs(pg, owner, repo);
+
+    for (const contrib of contribs) {
+      if ((await contributionExists(pg, contrib.id)).length !== 0) {
+        break;
+      }
+      if (await whitelistedContrib(pg, contrib, groups)) {
+        // insert to database
+        await addWhitelistedContrib(pg, contrib, proj.id);
+      }
+    }
+  });
+}
+
+function run() {
+  const options = {};
+
+  const pgp = pgPromise(options);
+  let pg = null;
+
+  // configure database connection
+  pg = pgp({
```

Comment:
Might want to read these from config as well

Reply:
Disagree, this is for the standalone script and should be fully independent of the service so it can be called via cron.

Reply:
This function is called when the script is executed as a standalone script, so it will not know what's in the config file, which is why I initialized the database connection specifically here.